SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -201,52 +201,109 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
+		FString CommandLineLocatorHost;
+		FParse::Value(FCommandLine::Get(), TEXT("locatorHost"), CommandLineLocatorHost);
+		if(!CommandLineLocatorHost.IsEmpty())
+		{
+			bShouldLoadFromURL = !Connection->LocatorConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = false;
+		}
+		else
+		{
+			bShouldLoadFromURL = !Connection->ReceptionistConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = true;
+		}
+
+		GameInstance->bFirstConnectionToSpatialOSAttempted = true;
 	}
-	else // Using Receptionist
-	{
-		Connection->ReceptionistConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
 
-		// Check for overrides in the travel URL.
-		if (!URL.Host.IsEmpty() && URL.Host.Compare(SpatialConstants::LOCAL_HOST) != 0)
+	// Copy relevant URL info into the relevant config object.
+	if(bShouldLoadFromURL)
+	{
+		if (URL.Host == SpatialConstants::LOCATOR_HOST || URL.HasOption(TEXT("locator")))
 		{
-			Connection->ReceptionistConfig.ReceptionistHost = URL.Host;
+			FLocatorConfig& LocatorConfig = Connection->LocatorConfig;
+			LocatorConfig.PlayerIdentityToken = URL.GetOption(*SpatialConstants::URL_PLAYER_IDENTITY_OPTION, TEXT(""));
+			LocatorConfig.LoginToken = URL.GetOption(*SpatialConstants::URL_LOGIN_OPTION, TEXT(""));
+			bUseReceptionist = false;
 		}
+		else
+		{
+			Connection->ReceptionistConfig.SetReceptionistHost(URL.Host);
+			bUseReceptionist = true;
+		}
+	}
 
-		bool bHasUseExternalIpOption = URL.HasOption(TEXT("useExternalIpForBridge"));
+	// Setup for the config objects regardless of loading from command line or URL
+	if (bUseReceptionist)
+	{
+		// Use Receptionist
+		Connection->SetConnectionType(ESpatialConnectionType::Receptionist);
+
+		FReceptionistConfig& ReceptionistConfig = Connection->ReceptionistConfig;
+		ReceptionistConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
 
-		if (bHasUseExternalIpOption)
+		const TCHAR* UseExternalIpForBridge = TEXT("useExternalIpForBridge");
+		if (URL.HasOption(UseExternalIpForBridge))
 		{
-			FString UseExternalIpOption = URL.GetOption(TEXT("useExternalIpForBridge"), TEXT(""));
-			if (UseExternalIpOption.Equals(TEXT("false"), ESearchCase::IgnoreCase))
-			{
-				Connection->ReceptionistConfig.UseExternalIp = false;
-			}
-			else
-			{
-				Connection->ReceptionistConfig.UseExternalIp = true;
-			}
+			FString UseExternalIpOption = URL.GetOption(UseExternalIpForBridge, TEXT(""));
+			ReceptionistConfig.UseExternalIp = !UseExternalIpOption.Equals(TEXT("false"), ESearchCase::IgnoreCase);
 		}
 	}
+	else
+	{
+		// Use Locator
+		Connection->SetConnectionType(ESpatialConnectionType::Locator);
+		FLocatorConfig& LocatorConfig = Connection->LocatorConfig;
+		FParse::Value(FCommandLine::Get(), TEXT("locatorHost"), LocatorConfig.LocatorHost);
+		LocatorConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
+	}
+
 
 	Connection->Connect(bConnectAsClient);
 }
 
+namespace
```

Comment:
nit: please add anonymous namespaces to the top of the file

Reply:
Done